src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1532,27 +1532,30 @@ public void ReInitializeLiveRunner()
         /// <param name="syncData"></param>
         public void ResetVMAndResyncGraph(IEnumerable<string> libraries)
         {
-            // Reset VM
-            ReInitializeLiveRunner();
-
-            if (!libraries.Any())
+            lock (mutexObject)
             {
-                return;
-            }
+                // Reset VM. This needs to be in mutex context as it turns DSExecutable null.
+                ReInitializeLiveRunner();
 
-            // generate import node for each library in input list
-            List<AssociativeNode> importNodes = new List<AssociativeNode>();
-            foreach (string lib in libraries)
-            {
-                ProtoCore.AST.AssociativeAST.ImportNode importNode = new ProtoCore.AST.AssociativeAST.ImportNode();
-                importNode.ModuleName = lib;
+                if (!libraries.Any())
+                {
+                    return;
+                }
 
-                importNodes.Add(importNode);
-            }
-            ProtoCore.CodeGenDS codeGen = new ProtoCore.CodeGenDS(importNodes);
-            string code = codeGen.GenerateCode();
+                // generate import node for each library in input list
+                List<AssociativeNode> importNodes = new List<AssociativeNode>();
+                foreach (string lib in libraries)
+                {
+                    ProtoCore.AST.AssociativeAST.ImportNode importNode = new ProtoCore.AST.AssociativeAST.ImportNode();
+                    importNode.ModuleName = lib;
 
-            UpdateCmdLineInterpreter(code);
```

Comment:
@mmisol does this mean we can get rid of this method? I see it is only used in tests apart from here and we could update the tests to use `SynchronizeInternal` directly.

Reply:
Since it's public I don't know if we can just do that, but we can definitely mark it as obsolete. I'll do that now.